features/support/multipart-post-2.0.0/lib/composite_io.rb

```diff
@@ -0,0 +1,108 @@
+#--
+# Copyright (c) 2007-2012 Nick Sieger.
+# See the file README.txt included with the distribution for
+# software license details.
+#++
+
+# Concatenate together multiple IO objects into a single, composite IO object
+# for purposes of reading as a single stream.
+#
+# Usage:
+#
+#     crio = CompositeReadIO.new(StringIO.new('one'), StringIO.new('two'), StringIO.new('three'))
+#     puts crio.read # => "onetwothree"
+#
+class CompositeReadIO
```

Comment:
Does adding this library to the `Gemfile` not work?

Reply:
It didn't work for me unflortunately. Don't think device farm considers the Gemfile at all and is pre loaded with some standard libs we are confined to.